frontend/src/employee-frontend/components/child-information/VasuAndLeops.tsx

```diff
@@ -213,16 +224,34 @@ const VasuAndLeops = React.memo(function VasuAndLeops({
               />
             </StateCell>
             <Td>
-              <InlineButton
-                onClick={() => history.push(`/vasu/${vasu.id}/edit`)}
-                text={i18n.common.edit}
-              />
+              {vasu.documentState === 'CLOSED' ? (
+                <RequireRole oneOf={['ADMIN']}>
+                  <InlineButton
+                    onClick={() =>
+                      void updateDocumentState({
+                        documentId: vasu.id,
+                        eventType: 'RETURNED_TO_REVIEWED'
+                      }).finally(() => loadVasus(childId))
+                    }
+                    text={i18n.vasu.transitions.RETURNED_TO_REVIEWED.buttonText}
+                  />
+                </RequireRole>
+              ) : (
+                <InlineButton
+                  onClick={() => history.push(`/vasu/${vasu.id}/edit`)}
+                  text={i18n.common.edit}
+                />
+              )}
             </Td>
           </Tr>
         ))
       )
       .getOrElse(null)
 
+  const allowCreation = vasus.isSuccess
+    ? vasus.value.find((doc) => doc.documentState !== 'CLOSED') === undefined
```

Comment:
I think `vasus.value.every(doc => doc.documentState === 'CLOSED')` communicates the intent better. 

Also, as more of a fyi, you could always use `vasus.map(...).getOrElse(...)` instead of checking for the state manually.